sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -106,21 +139,39 @@ public DigitalTwinsAsyncClient buildAsyncClient() {
         Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
         Objects.requireNonNull(endpoint, "'endpoint' cannot be null");
 
+        Configuration buildConfiguration = this.configuration;
+        if (buildConfiguration == null)
+        {
+            buildConfiguration = Configuration.getGlobalConfiguration().clone();
+        }
+
         // Set defaults for these fields if they were not set while building the client
-        this.serviceVersion = this.serviceVersion != null ? this.serviceVersion : DigitalTwinsServiceVersion.getLatest();
-        this.retryPolicy = this.retryPolicy != null ? this.retryPolicy : new RetryPolicy(); // Default is exponential backoff
+        DigitalTwinsServiceVersion serviceVersion = this.serviceVersion;
+        if (serviceVersion == null)
+        {
+            serviceVersion = DigitalTwinsServiceVersion.getLatest();
+        }
+
+        // Default is exponential backoff
+        HttpPipelinePolicy retryPolicy = this.retryPolicy;
+        if (retryPolicy == null)
+        {
+            retryPolicy = DEFAULT_RETRY_POLICY;
+        }
 
         if (this.httpPipeline == null) {
             this.httpPipeline = buildPipeline(
                 this.tokenCredential,
                 this.endpoint,
-                this.logOptions,
+                this.httpLogOptions,
                 this.httpClient,
                 this.additionalPolicies,
-                this.retryPolicy);
+                retryPolicy,
+                buildConfiguration,
+                this.properties);
         }
 
-        return new DigitalTwinsAsyncClient(this.httpPipeline, this.serviceVersion, this.endpoint);
+        return new DigitalTwinsAsyncClient(this.httpPipeline, serviceVersion, this.endpoint);
```

Comment:
I'm not a java expert but why was 'this' removed?

Reply:
In this case, removing the "this." means that I wanted to use the locally defined variable here instead of the instance variable. I've seen this pattern in the builders in track 2 where you don't assign the default value to the instance variables when building the pipeline. Instead, you create a local variable and assign its value to the default if the user never set the instance variable's value